pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -353,7 +371,23 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 				)
 			}).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			SkipIf:       v1beta1helper.GetShootServiceAccountKeyRotationPhase(o.Shoot.GetInfo().Status.Credentials) != gardencorev1beta1.RotationPreparing,
-			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerIsReady, waitUntilGardenerResourceManagerReady),
+			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerIsReady, waitUntilGardenerResourceManagerReady, waitUntilKubeAPIServerWithNodeAgentAuthorizerIsReady),
+		})
+		renewNodeAgentKubeconfigInAllNodes = g.Add(flow.Task{
+			Name: "Label nodes to trigger renewal of their gardener-node-agent kubeconfig",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return secretsrotation.RenewAccessSecretsInAllObjectsOfKind(ctx, o.Logger, o.ShootClientSet.Client(), &corev1.Node{}, v1beta1constants.GardenerOperationRenewKubeconfig)
+			}).RetryUntilTimeout(5*time.Second, 30*time.Second),
+			SkipIf:       !features.DefaultFeatureGate.Enabled(features.NodeAgentAuthorizer) || v1beta1helper.GetShootCARotationPhase(o.Shoot.GetInfo().Status.Credentials) != gardencorev1beta1.RotationPreparing,
+			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerIsReady, waitUntilGardenerResourceManagerReady, waitUntilKubeAPIServerWithNodeAgentAuthorizerIsReady),
+		})
+		_ = g.Add(flow.Task{
+			Name: "Check if all gardener-node-agents have finished the renewal of their kubeconfig",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return secretsrotation.CheckIfAccessSecretsRenewalCompletedInAllObjectsOfKind(ctx, o.ShootClientSet.Client(), &corev1.Node{}, v1beta1constants.GardenerOperationRenewKubeconfig)
+			}).RetryUntilTimeout(5*time.Second, 2*time.Minute),
+			SkipIf:       !features.DefaultFeatureGate.Enabled(features.NodeAgentAuthorizer) || v1beta1helper.GetShootCARotationPhase(o.Shoot.GetInfo().Status.Credentials) != gardencorev1beta1.RotationPreparing,
+			Dependencies: flow.NewTaskIDs(renewNodeAgentKubeconfigInAllNodes),
```

Comment:
Why do we need this? When the CA rotation is in `Preparing` phase, we roll out all nodes. New `node-agent`s (and `kubelet`s) come up with new client certs signed by the new client CA. Am I missing something?

Reply:
Yes, these tasks should not be needed unless we implement in-place node updates.

I'm not so sure about the node annotation. Without the annotation there would be no option to trigger the renewal of a GNA certificate manually. This could be helpful to debug issues like "the CSRs of my node xyz are not auto-approved".

Reply:
I don't think there is a real use-case for this. We don't have such functionality for kubelet certificates either.
If this indeed becomes necessary for in-place updates in the scope of #10219, it should be added separately for better traceability.